cmd/undo-main.go

```diff
@@ -215,20 +226,28 @@ func undoURL(ctx context.Context, aliasedURL string, last int, recursive, dryRun
 				break
 			}
 		}
-
 		if lastObjectPath != content.URL.Path {
 			// Print any object in the current list before reinitializing it
-			exitErr = undoLastNOperations(ctx, clnt, perObjectVersions, last, dryRun)
+			if remove {
+				exitErr = undoLastNOperations(ctx, clnt, perObjectVersions, last, dryRun)
+			}
+			remove = true
 			lastObjectPath = content.URL.Path
 			perObjectVersions = []*ClientContent{}
 		}
-
+		if !remove {
+			continue
+		}
+		if (content.IsLatest && action == "DELETE" && !content.IsDeleteMarker) || (content.IsLatest && action == "PUT" && content.IsDeleteMarker) {
```

Comment:
```suggestion
		if (content.IsLatest && action == "DELETE" && !content.IsDeleteMarker) || (content.IsLatest && action == "PUT" && content.IsDeleteMarker) {
```

Convert actions into constants such as 

```
const (
  actionPut    = "PUT"
  actionDelete = "DELETE"
)
```

It is generally preferred to have constants for strings like these.   actionPut = "PUT"